File: project/app/src/main/java/org/owntracks/android/injection/modules/ActivityModules/WelcomeActivityModule.java
package org.owntracks.android.injection.modules.ActivityModules;

import android.support.v7.app.AppCompatActivity;


import org.owntracks.android.injection.modules.ActivityModule;
import org.owntracks.android.injection.modules.FragmentModules.FinishFragmentModule;
import org.owntracks.android.injection.modules.FragmentModules.IntroFragmentModule;
import org.owntracks.android.injection.modules.FragmentModules.PermissionFragmentModule;
import org.owntracks.android.injection.modules.FragmentModules.PlayFragmentModule;
import org.owntracks.android.injection.modules.FragmentModules.VersionFragmentModule;
import org.owntracks.android.injection.scopes.PerActivity;
import org.owntracks.android.injection.scopes.PerFragment;
import org.owntracks.android.ui.welcome.WelcomeActivity;
import org.owntracks.android.ui.welcome.WelcomeMvvm;
import org.owntracks.android.ui.welcome.WelcomeViewModel;

import dagger.Binds;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module(includes = ActivityModule.class)
public abstract class WelcomeActivityModule {

    @Binds
    @PerActivity
    abstract AppCompatActivity bindActivity(WelcomeActivity a);

    @Binds
    @PerActivity
    abstract WelcomeMvvm.ViewModel bindViewModel(WelcomeViewModel viewModel);


    @ContributesAndroidInjector(modules = {PlayFragmentModule.class})
    @PerFragment
    abstract org.owntracks.android.ui.welcome.play.PlayFragment bindPlayFragment();

    @ContributesAndroidInjector(modules = {IntroFragmentModule.class})
    @PerFragment
    abstract org.owntracks.android.ui.welcome.intro.IntroFragment bindIntroFragment();

    @ContributesAndroidInjector(modules = {VersionFragmentModule.class})
    @PerFragment
    abstract org.owntracks.android.ui.welcome.version.VersionFragment bindVersionFragment();

    @ContributesAndroidInjector(modules = {PermissionFragmentModule.class})
    @PerFragment
    abstract org.owntracks.android.ui.welcome.permission.PermissionFragment bindPermissionFragment();

    @ContributesAndroidInjector(modules = {FinishFragmentModule.class})
    @PerFragment
    abstract org.owntracks.android.ui.welcome.finish.FinishFragment bindFinishFragment();
}

File: project/app/src/main/java/org/owntracks/android/ui/welcome/WelcomeActivityModule.kt
package org.owntracks.android.ui.welcome

import androidx.fragment.app.Fragment
import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.scopes.ActivityScoped
import dagger.hilt.components.SingletonComponent
import org.owntracks.android.ui.base.viewmodel.BaseViewModel
import org.owntracks.android.ui.welcome.finish.FinishFragment
import org.owntracks.android.ui.welcome.intro.IntroFragment
import org.owntracks.android.ui.welcome.permission.PermissionFragment
import org.owntracks.android.ui.welcome.permission.PlayFragment
import org.owntracks.android.ui.welcome.version.VersionFragment

@InstallIn(SingletonComponent::class)
@Module
abstract class WelcomeActivityModule {
//    @Binds
//    @ActivityScoped
//    abstract fun bindActivity(a: WelcomeActivity?): AppCompatActivity?

    @Binds
    @ActivityScoped
    abstract fun bindViewModel(viewModel: WelcomeViewModel?): BaseViewModel<WelcomeMvvm.View?>?

    @Binds
    @ActivityScoped
    abstract fun bindIntroFragment(introFragment: IntroFragment): Fragment

    @Binds
    @ActivityScoped
    abstract fun bindVersionFragment(versionFragment: VersionFragment): Fragment

    @Binds
    @ActivityScoped
    abstract fun bindPermissionFragment(permissionFragment: PermissionFragment): Fragment

    @Binds
    @ActivityScoped
    abstract fun bindFinishFragment(finishFragment: FinishFragment): Fragment

    @Binds
    @ActivityScoped
    abstract fun bindPlayFragment(playFragment: PlayFragment): Fragment
}